FakePrototypeStuff/JoystickInputToEvents.cs

```diff
@@ -21,6 +23,23 @@ public void Update()
 	public const int buttonCount = 20;
 	public const int joystickCount = 10;
 	private float[,] m_LastValues = new float[joystickCount, axisCount + buttonCount];
+
+	static string[] s_Axes;
+
+	static JoystickInputToEvents()
+	{
+		int first = 1;
+		int last = 10;
+		s_Axes = new string[(last - first) * joystickCount + 1];
+		for (int device = 0; device < joystickCount; device++)
```

Comment:
At the compiler level, back in the day, there was an advantage for pre-incrementing ints/counters in STL/For/etc loops. Unless post-increments no longer need/have copies made, and that is ALSO optimized at 1:1 parity, performance wise in C#/Mono, I vote that we switch to pre-incrementing(instead of post incrementing) wherever possible.

Reply:
Just looked into this. The assemblies are identical between a for loop with i++ vs ++i